src/main/scala/uk/gov/hmrc/agentclientauthorisation/model/SuspensionDetails.scala

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2022 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.agentclientauthorisation.model
+
+import play.api.libs.json.{Json, OFormat}
+import uk.gov.hmrc.agentclientauthorisation.model.Service._
+
+case class SuspensionDetails(suspensionStatus: Boolean, suspendedRegimes: Set[String]) {
+
+  def isRegimeSuspended(service: Service): Boolean = {
+    val regime = SuspensionDetails.serviceToRegime(service)
+    suspendedRegimes.contains(regime)
+  }
+
+  def isRegimeSuspended(id: String): Boolean = {
+    def idToService(id: String): Service = {
+      SuspensionDetails.serviceToRegime
+        .find(_._1.id == id)
+        .map(_._1)
+        .getOrElse(throw new IllegalArgumentException(s"Service of ID '$id' not known"))
+    }
+
+    val regime = SuspensionDetails.serviceToRegime(idToService(id))
+    suspendedRegimes.contains(regime)
+  }
+
+  def suspendedRegimesForServices(serviceIds: Set[String]): Set[String] = {
+    SuspensionDetails.serviceToRegime
+      .filterKeys(s => serviceIds.contains(s.id)).values.toSet
+      .intersect(suspendedRegimes)
+  }
+
+  def isAnyRegimeSuspendedForServices(ids: Set[String]): Boolean = suspendedRegimesForServices(ids).nonEmpty
+
+  override def toString: String = suspendedRegimes.toSeq.sorted.mkString(",")
+
+}
+
+object SuspensionDetails {
+
+  lazy val serviceToRegime: Map[Service, String] =
+    Map(MtdIt -> "ITSA", Vat -> "VATC", Trust -> "TRS", TrustNT -> "TRS", CapitalGains -> "CGT", PersonalIncomeRecord -> "PIR", Ppt -> "PPT")
```

Comment:
PIR is not a service that ETMP/DES handles so let's remove this please. An agent can't be suspended for PIR, at least not with our current setup.

Reply:
Removed PIR from `serviceToRegime` map, @geoffreywatson.